### PR TITLE
Move responsibility for managing GPS time stamp destination field to QgsProjectGpsSettings

### PR DESCRIPTION
**Pull request by nyalldawson**
and store time stamp destination fields in projects

Sponsored by NIWA

### CODE REVIEW

File: src/core/project/qgsprojectgpssettings.cpp [hunk 1]
```diff
@@ -85,6 +106,18 @@ QDomElement QgsProjectGpsSettings::writeXml( QDomDocument &doc, const QgsReadWri
     element.setAttribute( QStringLiteral( "destinationLayer" ), QString() );
   }
 
+  {
+    QDomElement timeStampElement = doc.createElement( QStringLiteral( "timeStampFields" ) );
```
Comment (nirvn) [inline src/core/project/qgsprojectgpssettings.cpp:110]: What about doing a bit of sanitation here and avoid saving details for a layer that's been removed (i.e. a layer ID that wouldn't match any project map layers anymore)?

**Review by nirvn — APPROVED** [review]
Looking good, just one suggestion to think about.

File: src/core/project/qgsprojectgpssettings.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/project/qgsprojectgpssettings.cpp:110]: Ok, done!

Comment (nirvn) [reply src/core/project/qgsprojectgpssettings.cpp:110]: @nyalldawson , thanks.

**Pull request merged**